tests/test_helpers/test_profiling.py

```diff
@@ -18,13 +18,40 @@
 #
 # ------------------------------------------------------------------------------
 """This module contains the tests for the helpers/profiling module."""
+import re
+from typing import Dict, List, Optional
+
+import pytest
+
 from aea.helpers.profiling import Profiling
 from aea.protocols.base import Message
 
 from tests.common.utils import wait_for_condition
 
 
-def test_profiling():
+def extract_object_counts(log: str) -> Dict[str, Dict[str, int]]:
+    """Extract object counts from the profiling log."""
+    result: Dict[str, Dict[str, int]] = {"created": {}, "present": {}}
+
+    for line in log.split("\n"):
+
+        match = re.match(r".*\* (?P<field>.*) \(present\):  (?P<count>\d+).*", line)
+        if match:
+            result["present"][match.groupdict()["field"]] = int(
+                match.groupdict()["count"]
+            )
+            continue
+
+        match = re.match(r".*\* (?P<field>.*) \(created\):  (?P<count>\d+).*", line)
+        if match:
+            result["created"][match.groupdict()["field"]] = int(
+                match.groupdict()["count"]
+            )
+
+    return result
```

Comment:
Helper function that extracts instance counts for present and created objects so it is easier and cleaner to check numbers.